core/network/src/protocol/specialization.rs

```diff
@@ -42,7 +42,8 @@ pub trait NetworkSpecialization<B: BlockT>: Send + Sync + 'static {
 	);
 
 	/// Called when a network-specific event arrives.
-	fn on_event(&mut self, event: Event);
+	#[deprecated(note = "This method is never called; please use `with_dht_event_tx` when building the service")]
+	fn on_event(&mut self, event: Event) {}
```

Comment:
Why not remove it completely? We don't have anything that uses it and we don't have anything released. Deprecating does not makes any sense here.

Reply:
Well, there's Polkadot. The idea was to then make a Polkadot PR, then later actually remove them.

Reply:
Hmm not sure. It is not that something updates Polkadot automatically. You could already open the pr to Polkadot and everybody would be happy.